lib/generators.nix

```diff
@@ -426,4 +426,59 @@ ${expr "" v}
       abort "generators.toDhall: cannot convert a null to Dhall"
     else
       builtins.toJSON v;
+
+  /*
+   * Translate a simple Nix expression to Lua representation with occasional
+   * Lua-inlines that can be construted by mkLuaInline function.
+   *
+   * generators.toLua {} {
+   *   cmd = [ "typescript-language-server" "--stdio" ];
+   *   settings.workspace.library = mkLuaInline ''vim.api.nvim_get_runtime_file("", true)'';
+   * }
+   *
+   *> {
+   *>   ["cmd"] = {
+   *>     "typescript-language-server",
+   *>     "--stdio"
+   *>   },
+   *>   ["settings"] = {
+   *>     ["workspace"] = {
+   *>       ["library"] = (vim.api.nvim_get_runtime_file("", true))
+   *>     }
+   *>   }
+   *> }
+   */
+  toLua = { indent ? "" }@args: v:
+    with builtins;
+    let
+      indent' = "${indent}  ";
```

Comment:
I find this confusing, can't the default for indent be `"  "` and remove `indent'` ?

Reply:
Unless I am mistaken, the initial `indent` is necessary for correctly indenting lists/attrsets.

Reply:
Later in code both `indent` (one passed in) and `indent'` (next level) are used. It is possible to replace all usages of `"${indent'}Something"` with `"${indent}  Something"`. But benefit of having it here allow to have only one place where we change indent. Of course we can use something like `"${indent}${indentInc}Something"`, but I find it way to much words there.  
Inner calls to `toLua` is done with `{ indent = indent'; }` to get inner values indented one level deeper.

I think I peeked into https://github.com/NixOS/nixpkgs/blob/4ec4c6fda90cec38b0878bf6e601f5df82ae5a8f/lib/generators.nix#L299-L300

I can change name(s) if main concern is use of tick (`'`) to indicate next level.

Reply:
Replaced use of ticks (`'`) with explicit names like `innerIndent` and `innerArgs`. Also fully incorporated part of `toPretty` related to indentation.